q2_assembly/tests/test_helpers.py

```diff
@@ -57,6 +61,132 @@ def test_is_valid_shortuuid(self):
+        )
+        genomes2 = GenomeSequencesDirectoryFormat(
+            self.get_data_path("genomes-dir-format2"), "r"
+        )
+        genomes = [genomes2, genomes1]
+        with self.assertRaises(TypeError):
+            assembly.methods.collate_genomes(genomes=genomes)
+
+    @parameterized.expand(["GenomeData", "DNAFASTAFormat"])
+    def test_collate_genomes_dnafastaformat_multiple_duplicates_warn(self, dir_fmt):
+        if dir_fmt == "GenomeData":
+            genomes1 = GenomeSequencesDirectoryFormat(
+                self.get_data_path("genomes-dir-format1"), "r"
+            )
+        else:
+            genomes1 = DNAFASTAFormat(
+                self.get_data_path("dna-fasta-format/dna-sequences1.fasta"), "r"
+            )
+        with warnings.catch_warnings(record=True) as w:
+            collated_genomes = collate_genomes(genomes=[genomes1, genomes1])
+            exp_files = ["ref1.fasta", "ref2.fasta"]
+            actual_files = sorted(os.listdir(collated_genomes.path))
+            self.assertEqual(actual_files, exp_files)
+            self.assertTrue(
+                any(
+                    "The latest occurrence will "
+                    "overwrite all previous" in str(msg.message)
+                    for msg in w
+                )
+            )
+
+            if dir_fmt == "DNAFASTAFormat":
+                content = {
+                    "ref1": {"description": "d_Bacteria_1", "sequence": "ACGTACGT"},
+                    "ref2": {"description": "d_Bacteria_2", "sequence": "CGTCGTCC"},
+                }
+
+                for fn in actual_files:
+                    fp = os.path.join(collated_genomes.path, fn)
+                    with open(fp, "r") as fasta_file:
+                        for seq in skbio.io.read(fasta_file, "fasta"):
+                            actual_id = seq.metadata["id"]
+                            actual_description = seq.metadata["description"]
+                            actual_sequence = str(seq)
+                            expected_id = fn.split(".")[0]
+                            expected_desc = content[expected_id]["description"]
+                            expected_sequence = content[expected_id]["sequence"]
+
+                            self.assertEquals(actual_id, expected_id)
+                            self.assertEqual(actual_description, expected_desc)
+                            self.assertEqual(actual_sequence, expected_sequence)
+
+    @parameterized.expand(["GenomeData", "DNAFASTAFormat"])
+    def test_collate_genomes_duplicates_error(self, dir_fmt):
+        if dir_fmt == "GenomeData":
+            genomes1 = GenomeSequencesDirectoryFormat(
+                self.get_data_path("genomes-dir-format1"), "r"
+            )
+        else:
+            genomes1 = DNAFASTAFormat(
+                self.get_data_path("dna-fasta-format/dna-sequences1.fasta"), "r"
+            )
+        with self.assertRaises(ValueError):
```

Comment:
We can make it better by using `assertRaisesRegex` and also asserting the error message itself.